crates/catalog/sql/src/catalog.rs

```diff
@@ -0,0 +1,517 @@
+            .metadata(metadata)
+            .build();
+
+        Ok(table)
+    }
+
+    async fn create_table(
+        &self,
+        namespace: &NamespaceIdent,
+        creation: TableCreation,
+    ) -> Result<Table> {
+        let location = creation.location.as_ref().ok_or(Error::new(
+            ErrorKind::DataInvalid,
+            "Table creation with the Sql catalog requires a location.",
+        ))?;
+        let name = creation.name.clone();
+
+        let uuid = Uuid::new_v4();
+        let metadata_location =
+            location.clone() + "/metadata/" + "0-" + &uuid.to_string() + ".metadata.json";
+
+        let metadata = TableMetadataBuilder::from_table_creation(creation)?.build()?;
+
+        let file = self.storage.new_output(&metadata_location)?;
+        file.writer()
+            .await?
+            .write_all(&serde_json::to_vec(&metadata)?)
+            .await?;
+        {
+            let catalog_name = self.name.clone();
+            let namespace = namespace.encode_in_url();
+            let name = name.clone();
+            let metadata_location = metadata_location.to_string();
+
+            sqlx::query(
+                &("insert into ".to_string()
+                    + CATALOG_TABLE_VIEW_NAME
+                    + " ("
+                    + CATALOG_NAME
+                    + ", "
+                    + TABLE_NAMESPACE
+                    + ", "
+                    + TABLE_NAME
+                    + ", "
+                    + METADATA_LOCATION_PROP
+                    + ") values (?, ?, ?, ?);"),
+            )
+            .bind(&catalog_name)
+            .bind(&namespace)
+            .bind(&name)
+            .bind(&metadata_location)
+            .execute(&self.connection)
+            .await
+            .map_err(from_sqlx_error)?;
+        }
+
+        Ok(Table::builder()
+            .file_io(self.storage.clone())
+            .metadata_location(metadata_location)
+            .identifier(TableIdent::new(namespace.clone(), name))
+            .metadata(metadata)
+            .build())
+    }
+
+    async fn rename_table(&self, _src: &TableIdent, _dest: &TableIdent) -> Result<()> {
+        todo!()
+    }
+
+    async fn update_table(&self, _commit: TableCommit) -> Result<Table> {
+        todo!()
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+    use iceberg::{
+        spec::{NestedField, PrimitiveType, Schema, Type},
+        Catalog, NamespaceIdent, TableCreation, TableIdent,
+    };
+    use tempfile::TempDir;
+
+    use crate::{SqlCatalog, SqlCatalogConfig};
+    use sqlx::migrate::MigrateDatabase;
+
+    #[tokio::test]
+    async fn test_create_update_drop_table() {
+        let dir = TempDir::with_prefix("sql-test").unwrap();
+        let warehouse_root = dir.path().to_str().unwrap();
+
+        //name of the database should be part of the url. usually for sqllite it creates or opens one if (.db found)
+        let sql_lite_url = "sqlite://iceberg";
```

Comment:
Is it possible to use in memory file or tmp file to make the tests reproducible?

Reply:
It automatically deletes the folder after running the test.